src/views/form.js

```diff
@@ -96,6 +96,30 @@ export const FormView = () => {
     })
   }, [isNavigateNext])
 
+  const [isNavigateSkip, setIsNavigateSkip] = useState(false)
+  useEffect(() => {
+    if (!isNavigateSkip || !question) return
+    const latestData = { ...data }
+    latestData[question.name] = null
+    setData(latestData)
+    storeFormData(latestData)
+    const effectPromise = question.effect
+      ? question.effect(latestData)
+      : Promise.resolve()
+    effectPromise.then(async (effectNextUrl) => {
+      if (effectNextUrl) {
+        history.push(effectNextUrl)
+      } else {
+        setIsFormVisible(false)
+        await waitSeconds(ANIMATION_TIME / 2000)
+        const nextUrl = getNextURL(url, Number(qIdx) + 1)
+        history.push(nextUrl)
+        setIsNavigateSkip(false)
+        setIsFormVisible(true)
+      }
+    })
+  }, [isNavigateSkip])
+
```

Comment:
this can probably be combined with the isNavigateNext effect rather than duplicating it
happy to discuss

Reply:
Implemented the fix in my local commit will push later.